apps/api/src/app/events/usecases/trigger-event/message-filter.matcher.ts

```diff
@@ -1,24 +1,23 @@
 import { NotificationStepEntity } from '@novu/dal';
+import * as _ from 'lodash';
 
-export function matchMessageWithFilters(
-  steps: NotificationStepEntity[],
-  payloadVariables: { [key: string]: string | string[] | { [key: string]: string } }
-): NotificationStepEntity[] {
-  return steps.filter((message) => {
-    if (message.filters?.length) {
-      const foundFilter = message.filters.find((filter) => {
-        if (filter.type === 'GROUP') {
-          return handleGroupFilters(filter, payloadVariables);
-        }
-
-        return false;
-      });
+export function matchMessageWithFilters(step: NotificationStepEntity, payloadVariables: any): boolean {
```

Comment:
```suggestion
export function matchMessageWithFilters(step: NotificationStepEntity, payloadVariables: any): boolean {
```

Can we add typing for the payload? The same type used by the API or Novu SDK